core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```
@@ -159,21 +159,26 @@ public PepperKitStatus syncKitStatusFromPepper(UUID kitId) {
     /**
      * Query Pepper for all in-progress kits and update the cached status in Juniper. This is intended to be called as a
      * scheduled job during expected non-busy times for DSM. If on-demand updates are needed outside the scheduled job,
-     * use {@link KitRequestService#updateKitStatus(AdminUser, UUID)} for a single kit or a batch operation that queries
+     * use {@link KitRequestService#syncKitStatusFromPepper} for a single kit or a batch operation that queries
      * Pepper for less than all open kits.
      */
     @Transactional
     public void syncAllKitStatusesFromPepper() {
         var studies = studyService.findAll();
         for (Study study : studies) {
             for (EnvironmentName environmentName : EnvironmentName.values()) {
-                syncKitStatusesForStudy(study, environmentName);
+                // if one sync fails, keep trying others in case the failure is just isolated unexpected data
+                try {
+                    syncKitStatusesForStudy(study, environmentName);
+                } catch (PepperParseException | PepperApiException e) {
+                    log.error("kit status sync failed for study %s".formatted(study.getShortcode()), e);
```

This probably deserves a comment, similar to the above, since swallowing exceptions should be considered exceptional (ahem).